src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAYLIST_SIZE = 15;
+    private int listSize;
+    private int lastRecord = 0;
+
+    private T[] listArray = (T[])(new Object[DEFAULT_ARRAYLIST_SIZE]);
```

Comment:
```suggestion
    private T[] array;
```
initialize in constructor